lib/cdo/slack.rb

```diff
@@ -83,8 +83,17 @@ def self.update_topic(channel_name, new_topic, use_channel_map = false)
       "&channel=#{channel_id}"\
       "&topic=#{new_topic}"
     )
+    result = JSON.parse(response.read)
+    error = result['error']
+    if error
+      if error == 'not_in_channel'
+        raise "Must join room #{channel_name} to be able to update topic"
+      else
+        raise "Failed to update topic, with error: #{result['error']}"
+      end
```

Comment:
The generic error of `Failed to update topic, with error: not_in_channel` did not seem super understandable without context, so I decided to special case that error with something more descriptive, and use the generic case for other things we might run into.